src/librbd/io/ImageRequest.cc

```diff
@@ -406,28 +406,6 @@ uint64_t ImageDiscardRequest<I>::append_journal_event(bool synchronous) {
   return tid;
 }
 
-template <typename I>
-int ImageDiscardRequest<I>::prune_object_extents(ObjectExtents &object_extents) {
```

Comment:
It looks like after this removal `prune_object_extents` is overridden only by `ImageCompareAndWriteRequest`, which actually does not do any prune but a validation. Probably rename the method to avoid confusion?